File: solutions/python/easy/p0217_contains_duplicate.py
"""LeetCode 217: Contains Duplicate.

Problem: https://leetcode.com/problems/contains-duplicate/
"""

from typing import List


class Solution:
    def containsDuplicate(self, nums: List[int]) -> bool:
        """Return whether any value appears more than once.

        Time: O(n)
        Space: O(n)
        """
        seen: set[int] = set()
        for number in nums:
            if number in seen:
                return True
            seen.add(number)
        return False


if __name__ == "__main__":
    solution = Solution()
    print(solution.containsDuplicate([1, 2, 3, 1]))
    print(solution.containsDuplicate([1, 2, 3, 4]))

File: solutions/java/easy/P0217_ContainsDuplicate.java
import java.util.HashSet;
import java.util.Set;

/**
 * LeetCode 217: Contains Duplicate.
 *
 * Problem: https://leetcode.com/problems/contains-duplicate/
 */
public class P0217_ContainsDuplicate {
    public boolean containsDuplicate(int[] nums) {
        // Time: O(n), Space: O(n)
        Set<Integer> seen = new HashSet<>();
        for (int number : nums) {
            if (!seen.add(number)) {
                return true;
            }
        }
        return false;
    }

    public static void main(String[] args) {
        P0217_ContainsDuplicate solution = new P0217_ContainsDuplicate();
        System.out.println(solution.containsDuplicate(new int[] {1, 2, 3, 1}));
        System.out.println(solution.containsDuplicate(new int[] {1, 2, 3, 4}));
    }
}
